ui/app/components/kv-object-editor.js

```diff
@@ -12,91 +20,59 @@
+ * @param {string} [keyPlaceholder] - placeholder for key input
+ * @param {string} [valuePlaceholder] - placeholder for value input
  */
 
-import { isNone } from '@ember/utils';
-import { assert } from '@ember/debug';
-import Component from '@ember/component';
-import { computed } from '@ember/object';
-import { guidFor } from '@ember/object/internals';
-import KVObject from 'vault/lib/kv-object';
-
-export default Component.extend({
-  'data-test-component': 'kv-object-editor',
-  classNames: ['field'],
-  classNameBindings: ['formSection:form-section'],
-  formSection: true,
-  // public API
-  // Ember Object to mutate
-  value: null,
-  label: null,
-  helpText: null,
-  subText: null,
-  // onChange will be called with the changed Value
-  onChange() {},
+export default class KvObjectEditor extends Component {
+  @tracked kvData;
 
-  init() {
-    this._super(...arguments);
-    const data = KVObject.create({ content: [] }).fromJSON(this.value);
-    this.set('kvData', data);
+  constructor() {
+    super(...arguments);
+    this.kvData = KVObject.create({ content: [] }).fromJSON(this.args.value);
     this.addRow();
-  },
-
-  kvData: null,
-
-  kvDataAsJSON: computed('kvData', 'kvData.[]', function () {
-    return this.kvData.toJSON();
-  }),
+  }
 
-  kvDataIsAdvanced: computed('kvData', 'kvData.[]', function () {
-    return this.kvData.isAdvanced();
-  }),
-
-  kvHasDuplicateKeys: computed('kvData', 'kvData.@each.name', function () {
-    let data = this.kvData;
-    return data.uniqBy('name').length !== data.get('length');
-  }),
+  get placeholders() {
+    return {
+      key: this.args.keyPlaceholder || 'key',
+      value: this.args.valuePlaceholder || 'value',
+    };
+  }
+  get hasDuplicateKeys() {
+    return this.kvData.uniqBy('name').length !== this.kvData.get('length');
+  }
 
+  @action
   addRow() {
-    let data = this.kvData;
-    let newObj = { name: '', value: '' };
-    if (!isNone(data.findBy('name', ''))) {
+    if (!isNone(this.kvData.findBy('name', ''))) {
       return;
     }
+    const newObj = { name: '', value: '' };
     guidFor(newObj);
-    data.addObject(newObj);
-  },
-  actions: {
-    addRow() {
-      this.addRow();
-    },
-
-    updateRow() {
-      let data = this.kvData;
-      this.onChange(data.toJSON());
-    },
-
-    deleteRow(object, index) {
-      let data = this.kvData;
-      let oldObj = data.objectAt(index);
-
-      assert('object guids match', guidFor(oldObj) === guidFor(object));
-      data.removeAt(index);
-      this.onChange(data.toJSON());
-    },
-
-    handleKeyUp(name, value) {
-      if (!this.onKeyUp) {
-        return;
-      }
-      this.onKeyUp(name, value);
-    },
-  },
-});
+    this.kvData.addObject(newObj);
+  }
+  @action
+  updateRow() {
+    this.args.onChange(this.kvData.toJSON());
+  }
+  @action
+  deleteRow(object, index) {
+    const oldObj = this.kvData.objectAt(index);
+    assert('object guids match', guidFor(oldObj) === guidFor(object));
```

Comment:
I left most of the logic as is when glimmerizing to avoid breaking anything but this doesn't seem useful to me. I think it would be fine to remove as well as the `guidFor(newObj)` on line 58.